openml/extensions/sklearn/extension.py

```diff
@@ -480,31 +480,163 @@ def _is_sklearn_flow(cls, flow: OpenMLFlow) -> bool:
     def _get_sklearn_description(self, model: Any, char_lim: int = 1024) -> str:
         '''Fetches the sklearn function docstring for the flow description
 
+        Retrieves the sklearn docstring available and does the following:
+        * If length of docstring <= char_lim, then returns the complete docstring
+        * Else, trims the docstring till it encounters a 'Read more in the :ref:'
+        * Or till it encounters a 'Parameters\n----------\n'
+        The final string returned is at most of length char_lim with leading and
+        trailing whitespaces removed.
+
         Parameters
         ----------
-        model: The sklearn model object
-        char_lim: int, specifying the max length of the returned string
+        model : sklearn model
+        char_lim : int
+            Specifying the max length of the returned string.
             OpenML servers have a constraint of 1024 characters for the 'description' field.
 
         Returns
         -------
-        string of length <= char_lim
+        str
         '''
         def match_format(s):
             return "{}\n{}\n".format(s, len(s) * '-')
         s = inspect.getdoc(model)
+        if s is None:
+            return ''
         if len(s) <= char_lim:
-            return s
+            # if the fetched docstring is smaller than char_lim, no trimming required
+            return s.strip()
         try:
-            pattern = "Read more in the :ref:"  # "Parameters"
+            # trim till 'Read more'
+            pattern = "Read more in the :ref:"
             index = s.index(pattern)
+            s = s[:index]
+            # trimming docstring to be within char_lim
+            if len(s) > char_lim:
+                s = "{}...".format(s[:char_lim - 3])
+            return s.strip()
         except ValueError:
+            pass
+        try:
+            # if 'Read more' doesn't exist, trim till 'Parameters'
             pattern = "Parameters"
             index = s.index(match_format(pattern))
+        except ValueError:
+            # returning full docstring
```

Comment:
Same as above, please don't hide errors.

Reply:
Added.

Reply:
Could you make this a warning, too, please?